app/models/ip_address.rb

```diff
@@ -0,0 +1,18 @@
+class IpAddress < ApplicationRecord
+  validates :ip_address, presence: true, uniqueness: true
+  validates :hashed_ip_address, presence: true, uniqueness: true
+
+  validates :geoip_info, presence: true, allow_nil: true
+  attribute :geoip_info, Types::JsonDeserializable.new(GeoipInfo)
+
+  before_validation :hash_ip_address!
+
+  has_many :user_events, class_name: "Events::UserEvent", dependent: :nullify
+  has_many :rubygem_events, class_name: "Events::RubygemEvent", dependent: :nullify
+
+  def hash_ip_address!
```

Comment:
What is reason to store hashed IP address?

Reply:
to allow for lookup of IPs from honeycomb, where we only store the hashed address